### PR TITLE
Add a setting to display large link previews

### PR DESCRIPTION
**Pull request by renchap**
Use-case and arguments for this can be found in #21874

This works similarly to the 16x9 setting for images.

image

The setting is currently enabled by default (I really like it better) but I can change it if you disagree.

When enabled, link previews will be a bigger card with a full-width media (styled inspired by @innosflew's [work here](https://github.com/mastodon/mastodon/issues/21874#issuecomment-1332556018)).

image

image

I chose to keep the host part immediately below the media as this image is much bigger and I feel giving the source of the link near the big clickable area it is a good idea. The text hierarchy works great like this as well.

This may need some adjustments, for example forcing an `aspect-ratio` on the preview (+ `object-fit: cover`), depending on feedback on more content. I will run with this for a few days on my own instance to see how it goes.

Preview image size might need to be made a little bigger as well, to 600px width maybe?

<details>
<summary>
If you want to test the styling part on your own instance, you can use this custom CSS
</summary>

```css
.status-card {
  background-color: #191b22;
  border-radius: 10px;
  flex-direction: column;
}
.status-card .status-card .status-card__image {
  flex: unset;
}
.status-card .status-card .status-card__image .status-card .status-card__image-image {
  border-radius: 0;
}
.status-card .status-card__content {
  padding: 15px;
  display: flex;
  flex-direction: column;
}
.status-card .status-card__host, .status-card .status-card__description {
  color: #606984;
}
.status-card .status-card__title {
  order: 2;
  margin-bottom: 0;
}
.status-card .status-card__host {
  order: 1;
}
.status-card .status-card__description {
  order: 3;
}
```

</detail>

### CODE REVIEW

**Comment by ClearlyClaire** [comment]
It looks good overall, although a bit large, and I'm not sure about the padding/spacing around the “host”/“provider” part. It would also be good to double-check for consistency with the existing similar layout in expanded status view.

File: app/javascript/mastodon/features/status/components/card.js [hunk 1]
```diff
@@ -194,7 +194,7 @@ export default class Card extends React.PureComponent {
     const provider    = card.get('provider_name').length === 0 ? decodeIDNA(getHostname(card.get('url'))) : card.get('provider_name');
     const horizontal  = (!compact && card.get('width') > card.get('height') && (card.get('width') + 100 >= width)) || card.get('type') !== 'link' || embedded;
     const interactive = card.get('type') !== 'link';
-    const className   = classnames('status-card', { horizontal, compact, interactive });
+    const className   = classnames('status-card', { horizontal, compact, interactive, large: displayLargePreviews });
```
Comment (nyura123) [inline app/javascript/mastodon/features/status/components/card.js:197]: should `large` be set to true only if `card.image` is truthy, to handle links without images? For ex. https://mstdn.social/@meg@dair-community.social/109439708179854100

**Comment by afontenot** [comment]
This doesn't appear to change the size of the image generated for the preview, so it will be upscaled. I have a comment here with a fix for that: https://github.com/mastodon/mastodon/issues/19984#issuecomment-1328374100

I also agree that this is a bit large looking. I wonder if limiting the height slightly might make it work better on smaller screens.

**Comment by Gargron** [comment]
Rather than making this a setting, I'd be OK with making it the new default. Alternatively, connecting it to the OpenGraph setting of the page (summary vs summary_large_image).

**Comment by Erikcht** [comment]
Why does thumb work on mobile and not on desktop browser?

**Comment by ronilaukkarinen** [comment]
I wonder why did this PR stall? I've been using a code based on [this comment](https://github.com/mastodon/mastodon/issues/19984#issuecomment-1328066798) and it works brilliantly. If it doesn't need a setting in @Gargron's opinion and could be the new default, isn't a new PR in place?

**Comment by j-f1** [comment]
The iOS app does this too, and the logic that it currently uses is that it sticks with the horizontal layout if the image is nearly square (or missing entirely), and uses the large vertical layout if the image is not square.

**Comment by Erikcht** [comment]
Are the thumbs with youtube, via browser, working in this test version?
Do you have any youtube url demos?

**Comment by renchap** [comment]
Closing as it has been implemented in https://github.com/mastodon/mastodon/pull/26136

**Pull request closed without merging**